android/src/main/java/com/marigold/rnsdk/RNMessageStreamModule.kt

```diff
@@ -27,20 +30,56 @@ class RNMessageStreamModule (reactContext: ReactApplicationContext, private val
     @VisibleForTesting
     internal var jsonConverter: JsonConverter = JsonConverter()
 
+    private val eventChannel = Channel<Boolean>()
+    private var defaultInAppNotification = displayInAppNotifications
     init {
         messageStream.setOnInAppNotificationDisplayListener(this)
     }
 
     override fun shouldPresentInAppNotification(message: Message): Boolean {
-        try {
-            val writableMap = jsonConverter.convertJsonToMap(message.toJSON())
-            reactApplicationContext
+        if (defaultInAppNotification) {
+            return true
+        }
+
+        return runBlocking {
+            val result = async { emitWithTimeout(message) }
```

Comment:
What's the reason for the async block here? Can we not just call the method directly for the result?